src/device.cpp

```diff
@@ -1196,11 +1196,15 @@ cl_int cvk_device::get_device_host_timer(cl_ulong* device_timestamp,
     return CL_SUCCESS;
 }
 
-cl_ulong cvk_device::device_timer_to_host(cl_ulong dev, cl_ulong sync_dev,
-                                          cl_ulong sync_host) const {
-    if (sync_host > sync_dev) {
-        return (sync_host - sync_dev) + dev;
+cl_ulong cvk_device::device_timer_to_host(cl_ulong dev) {
+    if (dev > m_sync_dev) {
```

Comment:
```suggestion
    if (dev <= m_sync_dev) {
```
Isn't your condition backwards? You want a new pair of aligned timestamps only when you've wrapped around and the timestamp to convert to the host's time base is smaller than the device timestamp used as a reference.

Reply:
I don't think it is backwards.
The idea is to get new sync points when the time we want is bigger than what we have converted so far. Because they might have been a slight divergence since we have the last sync point.

Reply:
So you'll end up requesting a new pair of timestamps for pretty much each command since the device timestamp will almost always be greater than the last one obtained when getting a host/device pair. The only time where you're not requesting a host/device pair is when the device timestamp wraps around. At which point I don't see much motivation for the change.